src/main/java/utilities/Matrix.java

```diff
@@ -0,0 +1,140 @@
+package utilities;
+

```

Comment:
Admittedly, the matrix class is a bit confusing and complicated and I do not think it is good enough, either. But I do not have a good way to improve it. First, it should be noted that an Id may not always be an integer. Some users may prefer to use String (e.g., region name) to declare an Id.  For example, I used "region01" in my test. So I hope to develop a more general matrix class. String tends to be more general, as Integer can be converted to String.
Yes, we can use bi-dimensional array, but we may also need another matrix to link the array index to object IDs. For example, given that travelTime[1,2] stores the travel time 
 between region0 and region3, you need to firstly find out the row and column indices ([1,2]) in another array, say regionID2TravelTimeIndices [region0, region3], with the region Ids ([region0, region3]). This is also complicated. Do you have any other suggestions?

Reply:
I understand and agree that there is a problem here, but I think the problem is precisely created because of using ID class... as in, the use of the ID class imposes on you the use of some kind of Matrix class, simply because ids can be non-integers. If ids are kept as integers, then a bi-dimensional array solves the problem. Also, no matter what the ids are, can you not solve it with a hashMap of hashMaps?

Reply:
I will try to use integers instead to describe the IDs, which I think should be easier to understand at the moment, especially for core developers of this project, as you mentioned above that there might not be too many external users in the near term.